### PR TITLE
Refactoring validator

### PR DESCRIPTION
**Pull request by Gallardot**
### What problem does this PR solve? 
#291
### What is changed and how does it work?

use `webhook.Validator` to refactoring validator
unity the form of webhook
add Unit test for validator

### Check List 

Tests 

 - Unit test

Code changes

 - Has Go code change

Side effects

 - 

Related changes

 - 

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by cwen0** [comment]
@Gallardot Good job!

**Comment by YangKeao** [comment]
It seems that I cannot delete a chaos resource. Do you know what the problem is ?

```
Error from server: admission webhook "vtimechaos.kb.io" denied the request: there is no content to decode
```

**Comment by YangKeao** [comment]
It seems related with this [PR](https://github.com/kubernetes/kubernetes/pull/76346) on kubernetes, which means Validator will break compatibility with old version of kubernetes.

The former error occurs on kubernetes 1.14.8. As I have tested, 1.17.0 works well.

**Comment by cwen0** [comment]
> It seems related with this [PR](https://github.com/kubernetes/kubernetes/pull/76346) on kubernetes, which means Validator will break compatibility with old version of kubernetes. There is also a related [issue](https://github.com/kubernetes-sigs/controller-runtime/issues/845) for controller-runtime.
> 
> The former error occurs on kubernetes 1.14.8. As I have tested, 1.17.0 works well.

About this issue, Can we disable validator by default? When the controller-runtime fix this https://github.com/kubernetes-sigs/controller-runtime/issues/845, we enable validator.

**Comment by Gallardot** [comment]
> It seems related with this [PR](https://github.com/kubernetes/kubernetes/pull/76346) on kubernetes, which means Validator will break compatibility with old version of kubernetes. There is also a related [issue](https://github.com/kubernetes-sigs/controller-runtime/issues/845) for controller-runtime.
> 
> The former error occurs on kubernetes 1.14.8. As I have tested, 1.17.0 works well.

Thank you. I will test it on kubernetes 1.14.8 later. Maybe we can ignore the `DELETE` operations. I will have a try.

**Comment by cwen0** [comment]
@Gallardot Hi, Currently, we are preparing Chaos Mesh-for CNCF Sandbox entry. https://github.com/cncf/toc/pull/367. At present, we are preparing the slide which requires some contributor-related information.  Would you mind tell us what organization are you from?

**Comment by Gallardot** [comment]
@YangKeao @cwen0 

I reproduce this error on kubernetes 1.14.8. Currently, we do nothing for `DELETE` operations. Maybe we can ignore the `DELETE ` operations on the `ValidatingWebhookConfiguration`. It is Ok on kubernetes 1.14.8. Please take a look.

File: api/v1alpha1/common_webhook.go [hunk 1]
```diff
@@ -13,9 +13,33 @@
...
 // DefaultNamespace set the namespace of chaos object as the default namespace selector if namespaces not set
 func (in *SelectorSpec) DefaultNamespace(namespace string) {
 	if len(in.Namespaces) == 0 {
 		in.Namespaces = []string{namespace}
 	}
 }
+
+// +kubebuilder:object:generate=false
+
+// ChaosValidator describe the interface should be implemented in chaos
```
Comment (cwen0) [inline api/v1alpha1/common_webhook.go]: s/describe/describes/g

File: api/v1alpha1/common_webhook.go [hunk 2]
```diff
@@ -13,9 +13,33 @@
...
+type ChaosValidator interface {
+	webhook.Validator
+	// Validate validate chaos object
```
Comment (cwen0) [inline api/v1alpha1/common_webhook.go]: s/validate/validates/g

File: api/v1alpha1/common_webhook.go [hunk 3]
```diff
@@ -13,9 +13,33 @@
...
+	Validate() error
+	// ValidateScheduler validate the scheduler and duration
```
Comment (cwen0) [inline api/v1alpha1/common_webhook.go]: ditto, and please fix the same issue at the following code.

Comment (Gallardot) [reply api/v1alpha1/common_webhook.go]: Thank you. All done.Please take a look.

**Pull request merged**